src/manifests_workflow/input_manifests_opensearch.py

```diff
@@ -0,0 +1,99 @@
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import subprocess
+
+from manifests.input_manifest import InputManifest
+from manifests_workflow.component_opensearch import ComponentOpenSearch
+from manifests_workflow.component_opensearch_min import ComponentOpenSearchMin
+from manifests_workflow.input_manifests import InputManifests
+from system.temporary_directory import TemporaryDirectory
+
+
+class InputManifestsOpenSearch(InputManifests):
+    @classmethod
+    def files(self):
+        return InputManifests.files("opensearch")
+
+    def update(self, keep=False):
+        known_versions = self.versions
+        logging.info(f"Known versions: {known_versions}")
+        main_versions = {}
+        with TemporaryDirectory(keep=keep) as work_dir:
+            logging.info(f"Checking out components into {work_dir}")
+            os.chdir(work_dir)
+
+            # check out and build OpenSearch#main, 1.x, etc.
+            branches = ComponentOpenSearchMin.branches()
+            logging.info(f"Checking OpenSearch {branches} branches")
+            for branch in branches:
+                c = ComponentOpenSearchMin.checkout(
+                    path=os.path.join(work_dir, f"OpenSearch/{branch}"), branch=branch
+                )
+                c.publish_to_maven_local()
+                version = c.version
+                logging.info(f"OpenSearch#{branch} is version {version}")
+                if version not in main_versions.keys():
+                    main_versions[version] = [c]
+
+            # components can increment their own version first without incrementing min
+            manifest = self.latest
+            for component in manifest.components:
+                if component.name == "OpenSearch":
+                    continue
+
+                logging.info(f"Checking out {component.name}#main")
+                component = ComponentOpenSearch.checkout(
+                    name=component.name,
+                    path=os.path.join(work_dir, component.name),
+                    version=manifest.build.version,
+                    branch="main",
+                )
+
+                try:
+                    component_version = component.version
+                    if component_version:
+                        release_version = ".".join(component_version.split(".")[:3])
+                        if release_version not in main_versions.keys():
+                            main_versions[release_version] = []
+                        main_versions[release_version].append(component)
+                        logging.info(
+                            f"{component.name}#main is version {release_version} (from {component_version})"
+                        )
+                except subprocess.CalledProcessError as err:
+                    logging.warn(
+                        f"Error getting version of {component.name}: {str(err)}, ignored"
```

Comment:
This happen enough that its worth swallowing? Seems like we would want to error out